tests/mock_pytket_backend.py

```diff
@@ -0,0 +1,82 @@
+from typing import Optional, Union, List, Sequence, Set, cast
```

Comment:
Why are you defining this in the tests directory?

Reply:
Do you mean why the``MockShotBackend`` is defined in tests?

Reply:
The `MockShotBackend` is only used for testing qiskit embedding. I'm not sure where else I can define it.

Reply:
At least for now, I don't think this backend can be useful elsewhere. These methods return fake data and can cause issues if the users aren't aware of the implementation (e.g. the number of shots returned from the get_result and the number of shots requested don't match). In the future, I think it might make more sense to have a configurable ``TketSimBackend`` (with correct behaviours) defined in pytket that can be used to test various things.